acceptance-tests/src/test/java/tech/pegasys/eth2signer/dsl/signer/Eth2SignerProcessRunner.java

```diff
@@ -0,0 +1,230 @@
+            "Failed to create the temporary directory to store the eth2signer.ports file");
+      }
+    } else {
+      dataPath = null;
+    }
+  }
+
+  @SuppressWarnings("UnstableApiUsage")
+  public synchronized void shutdown() {
+    final HashMap<String, Process> localMap = new HashMap<>(processes);
+    localMap.forEach(this::killProcess);
+    outputProcessorExecutor.shutdown();
+    try {
+      if (!outputProcessorExecutor.awaitTermination(5, TimeUnit.SECONDS)) {
+        LOG.error("Output processor executor did not shutdown cleanly.");
+      }
+    } catch (final InterruptedException e) {
+      LOG.error("Interrupted while already shutting down", e);
+      Thread.currentThread().interrupt();
+    } finally {
+      if (useDynamicPortAllocation) {
+        try {
+          MoreFiles.deleteRecursively(dataPath, RecursiveDeleteOption.ALLOW_INSECURE);
+        } catch (final IOException e) {
+          LOG.info("Failed to clean up temporary file: {}", dataPath, e);
+        }
+      }
+    }
+  }
+
+  public void start(final String processName) {
+    final String loggingLevel = "DEBUG";
+
+    final List<String> params = new ArrayList<>();
+    params.add(executableLocation());
+    params.add("--logging");
+    params.add(loggingLevel);
+    params.add("--http-listen-host");
+    params.add(signerHostname);
+    params.add("--http-listen-port");
+    params.add(String.valueOf(signerHttpRpcPort));
+    params.add("--key-store-path");
+    params.add(signerConfig.getKeyStorePath().toString());
+    if (useDynamicPortAllocation) {
+      params.add("--data-path");
+      params.add(dataPath.toAbsolutePath().toString());
+    }
+
+    final StringJoiner javaOpts = new StringJoiner(" ");
+
+    LOG.info("Creating Eth2Signer process with params {}", params);
+
+    final ProcessBuilder processBuilder =
+        new ProcessBuilder(params)
+            .directory(new File(System.getProperty("user.dir")).getParentFile())
+            .redirectErrorStream(true)
+            .redirectInput(Redirect.INHERIT);
+
+    if (Boolean.getBoolean("debugSubProcess")) {
+      javaOpts.add("-Xdebug -agentlib:jdwp=transport=dt_socket,server=y,suspend=y,address=5005");
+    }
+    processBuilder.environment().put("JAVA_OPTS", javaOpts.toString());
+
+    try {
+      final Process process = processBuilder.start();
+      outputProcessorExecutor.submit(() -> printOutput(processName, process));
+      processes.put(processName, process);
+    } catch (final IOException e) {
+      LOG.error("Error starting EthSigner process", e);
+      throw new RuntimeException("Failed to start the Ethsigner process");
+    }
+
+    if (useDynamicPortAllocation) {
+      loadPortsFile();
+    }
+  }
+
+  public boolean isRunning(final String processName) {
+    return (processes.get(processName) != null) && processes.get(processName).isAlive();
+  }
+
+  private String executableLocation() {
+    return "build/install/eth2signer/bin/eth2signer";
```

Comment:
This originally defines in gradle, correct? I wonder if there is a way to inject via system property and then default to this standard location ... or am I overthinking.

Reply:
_personally_ I think overthinking - just replicating EthSigner and Besu

Reply:
👍